packages/tailwindcss/src/variants.test.ts

```diff
@@ -1816,12 +1841,23 @@ test('data', () => {
     run([
       'data-disabled:flex',
       'data-[potato=salad]:flex',
+      'data-[foo=1]:flex',
+      'data-[foo=bar baz]:flex',
+      "data-[foo$='bar' i]:flex",
```

Comment:
We should use `_` instead of ` ` here:
```suggestion
      'data-[foo=bar_baz]:flex',
      "data-[foo$='bar'_i]:flex",
```